protocol-tokio/src/protocol/codec/message.rs

```diff
@@ -240,11 +252,11 @@ impl<T: de::Deserialize, E: de::Deserialize> de::Deserialize for Response<T, E>
 }
 
 #[derive(Debug, PartialEq, Eq, Clone)]
-pub struct GetBlockHeaders {
-    pub from: Vec<HeaderHash>,
-    pub to: Option<HeaderHash>,
+pub struct GetBlockHeaders<T> {
+    pub from: Vec<T>,
+    pub to: Option<T>,
 }
-impl GetBlockHeaders {
+impl<T> GetBlockHeaders<T> {
```

Comment:
It might make sense to restrict the type parameter to the chain-core classifier trait `property::BlockId`, to prevent accidental parameterization of `GetBlockHeader` with unintended types.